blocks/json/json.h

```diff
@@ -162,17 +162,13 @@ CURRENT_STRUCT(JSONObject) {
   JSONObject& push_back(const std::string& name, const JSONValue& value) {
     auto it_field = fields.find(name);
     if (it_field == std::end(fields)) {
-      keys.push_back(name);
       fields.emplace(name, value);
+      keys.push_back(name);
     } else {
-      auto it_list = std::find(std::begin(keys), std::end(keys), name);
-      auto it_list_next = it_list;
-      while (++it_list_next != std::end(keys)) {
-        *it_list = *it_list_next;
-        ++it_list;
-      }
-      *it_list = name;
       it_field->second = value;
+      auto it_list = std::find(std::begin(keys), std::end(keys), name);
+      std::rotate(it_list, it_list + 1u, std::end(keys));
+      keys.back() = name;
```

Comment:
This line seems redundant. Feel free to merge after fixing :)

Reply:
:+1: